src/FunctionalArrayCollection.js

```diff
@@ -0,0 +1,69 @@
+import FunctionalCollection from './FunctionalCollection'
+
+/**
+ * Functional collection that works with arrays under the hood
+ */
+export class FunctionalArrayCollection extends FunctionalCollection {
+    /**
+     * Retrieves items of the collection.
+     *
+     * @return {Array} The items, by item ID.
+     */
+    getItems () {
+        return this._getItems().slice()
+    }
+
+    /**
+     * Check that given item in the collection.
+     *
+     * @param item
+     * @returns {boolean}
+     */
+    hasItem (item) {
+        let foundItemIndex = null
+        let items = this.getItems()
+        for (let i in items) {
+            if (this._keyGetter(items[i]) === this._keyGetter(item)) {
```

Comment:
You are invoking the `_keyGetter()` method multiple times for the same target items - as many times as there are items in the collection. I'm pretty sure you can extract this out of the `for..in` block.

Reply:
👍